cmd/common/image/inspect.go

```diff
@@ -49,6 +59,62 @@ func NewInspectCmd() *cobra.Command {
 				return fmt.Errorf("inspecting image: %w", err)
 			}
 
+			runtimeGlobalParams := runtime.GlobalParamDescs().ToParams()
+			runtimeParams := runtime.ParamDescs().ToParams()
+			ociParams := apihelpers.ToParamDescs(ocihandler.OciHandler.InstanceParams()).ToParams()
+
+			// Add operator global flags
+			opGlobalParams := make(map[string]*params.Params)
+			for _, op := range operators.GetDataOperators() {
+				opGlobalParams[op.Name()] = apihelpers.ToParamDescs(op.GlobalParams()).ToParams()
+			}
+
+			var info *api.GadgetInfo
+			err = runtime.Init(runtimeGlobalParams)
+			if err != nil {
+				return fmt.Errorf("initializing runtime: %w", err)
+			}
+			defer runtime.Close()
+
+			// set global operator flags from the config file
+			for o, p := range opGlobalParams {
+				err = common.SetFlagsForParams(cmd, p, config.OperatorKey+"."+o)
+				if err != nil {
+					return fmt.Errorf("setting operator %s flags: %w", o, err)
+				}
+			}
+
+			ops := make([]operators.DataOperator, 0)
+			for _, op := range operators.GetDataOperators() {
+				// Initialize operator
+				err := op.Init(opGlobalParams[op.Name()])
+				if err != nil {
+					continue
+				}
+				ops = append(ops, op)
+			}
+
+			gadgetCtx := gadgetcontext.New(
+				context.Background(),
+				image.String(),
+				gadgetcontext.WithDataOperators(ops...),
+				gadgetcontext.WithUseInstance(false),
+				gadgetcontext.IncludeExtraInfo(true),
+			)
+
+			paramValueMap := make(map[string]string)
+			ociParams.CopyToMap(paramValueMap, "operator.oci.")
+
+			info, err = runtime.GetGadgetInfo(gadgetCtx, runtimeParams, paramValueMap)
+			if err != nil {
+				return fmt.Errorf("getting gadget info: %w", err)
+			}
+			fmt.Println(info.ExtraInfo.Data["ebpf.sections"])
+			fmt.Println(info.ExtraInfo.Data["ebpf.maps"])
+			fmt.Println(info.ExtraInfo.Data["ebpf.programs"])
+			fmt.Println(info.ExtraInfo.Data["wasm.upcalls"])
+			fmt.Println(info.ExtraInfo.Data["wasm.gadgetAPIVersion"])
```

Comment:
Can you iterate over `Data` in a generic way and insert it in the json/jsonpretty/yaml?

For the column output, I am thinking we would not print the information since it would not fit.

We could also have a custom output for showing one section. For example:
```
$ sudo ig image inspect myimage -o custom --extrainfo=ebpf.programs
```